src/sync/writer.rs

```diff
@@ -113,13 +115,16 @@ pub(super) struct DataSyncCollection {
 // An object corresponding to a single `do_put` call.
 #[derive(Debug, Clone, PartialEq)]
 pub(super) struct DataSyncItem {
+    // Flag denoting whether the batches have been physically squashed yet or not. If true there is
+    // a single record batch in the `data` field, and this is strictly the case during flushing.
+    pub(super) squashed: bool,
```

Comment:
```suggestion
    pub(super) has_been_squashed: bool,
```
or `is_squashed`. Because then it shows that the batches have been changed, and not is a flag allowing it.